src/test/java/org/jabref/gui/search/GlobalSearchBarTest.java

```diff
@@ -88,8 +89,8 @@ void recordingSearchQueriesOnFocusLostOnly(FxRobot robot) throws InterruptedExce
 
         // Set the focus to another node to trigger the listener and finally record the query.
         DefaultTaskExecutor.runInJavaFXThread(() -> hBox.requestFocus());
-        List<String> lastSearchHistory = stateManager.getWholeSearchHistory();
-        List<String> expected = List.of("Smith");
```

Comment:
You can keep the List interface. The test failure was misleading. It turned out that it was empty:
you can add an assertFalse(lastSearchHistory.isEmpty())

Reply:
Works fine!